Compass documentation (#9035)
mendix/docs: content/en/docs/developerportal/compass.md

---

> title: "Launching your First Mendix App with Compass"

Reviewer: Is this really about launching an app? It seems to be more about preparing your organization for the first Mendix app. It is much more waterfall than the normal Agile way of creating a first app.

I wonder if we need a title which is more indicative of being part of a large organization.

> linktitle: "Compass"

Reviewer: Is this the best title? It implies that customers already know what Compass is and how it can help them. I'm not sure if most organizations will be looking for "Compass" - but perhaps they find this through other channels like Marketing and Blogs?

url: /developerportal/compass/
weight: 25
description: "Describes the Compass tool."
---

{{% alert color="info" %}} This feature is currently available to participating customers. For more information, contact your Customer Service Manager. {{% /alert %}}

## Introduction

> The [Compass](https://compass.home.mendix.com/) tool is available to new Mendix users who want to plan and streamline their implementation tasks. It is part of an initiative to bring [Mendix Digital Execution](https://www.mendix.com/glossary/digital-execution/) to the Platform, and enables non-developer team members to help plan the milestones and goals of their Mendix implementation process. The Program Owner (that is, the team member leading the implementation) can then monitor the progress through an intuitive visual interface, with the goal of successfully launching their first Mendix application.

Reviewer: Not clear who the reader of this is. Who is the "you" in the rest of the document. Is it a "new Mendix user", a "non-developer team member", or "The Program Owner"? Who is the "their" in "successfully launching their first Mendix application"?

From the way the functions are described, I think it is probably "The Program Owner" and I think we should rewrite the introduction from their point of view.

{{< figure src="/attachments/developerportal/compass/compass-home.png" alt="The home page of the Compass tool" >}}

> With the Compass tool, you can define your own vision of low-code and the apps that you want to build with Mendix, and then simplify their implementation by following a number of predefined best-practice milestones. Each of those milestones consists of action points which you can assign to members of your team for easy tracking. If required, you can also define additional tasks for each milestone.

Reviewer: "your team" - is this the same team as is identified in https://docs.mendix.com/developerportal/general/team/, or is it some other team. If it is the app team, then perhaps put a link here?

## Getting Started with Compass

> To enable Compass for your Mendix implementation, contact your Mendix Customer Service Manager. 

Reviewer: To match the description two paragraphs later, I think this should be
```suggestion
To enable Compass for your organization, contact your Mendix Customer Service Manager. 
```

{{% alert color="info" %}}

> Compass is a tool intended for new users of Mendix who want to launch their first application. Many of the milestones and tasks are geared towards beginners, so teams which have already created one or more Mendix apps are less likely to find them useful.

Reviewer: Is this tool for _new users_, _new teams_ ,or _organizations which are new to Mendix_. I don't think we are being clear.

{{% /alert %}}

> After Compass has been enabled for your organization, the Program Owner can define the launch date and invite team members to participate.

Reviewer: How do they define the launch date?
Is there one "Compass" per app or one "Compass" per organization.

### Inviting Team Members

You can give members of your team access to the tool by performing the following steps:

> 1. Log in to [Compass](https://compass.home.mendix.com/) as a user with the Program Owner role.

Reviewer: We talk about "the Program Owner" earlier, but this implies that there could be multiple Program Owners? I think we should align this.

2. In the left navigation menu, click **Access Management**.
3. In the **Invite Members** window, enter the email address of the user that you want to invite, and specify their role in the project.

{{% alert color="info" %}}

> The Program Owner role grants users access to inviting other users. As a best practice, follow the principle of least privilege, and only assign this role to project leaders.

Reviewer: We are introducing a new term here "project leaders". I think we should stick to the term we have chosen "Program Owner" which we already define above as "the team member leading the implementation". Otherwise, we have to explain what we mean by "project leader".

```suggestion
The Program Owner role grants users access to invite other users. As a best practice, follow the principle of least privilege, and only assign this role to team members who are leading the implementation.
```

{{% /alert %}}

You can invite users who are already a part of your organization, but also external users, for example, from partner organizations. Their invitations are displayed in the **Pending Invitations** tab until accepted.

{{< figure src="/attachments/developerportal/compass/invite.png" alt="A screenshot of the Invite Members window" >}}
  
## Tracking Milestones and Action Points

Milestones are key points in the implementation process that must be completed in order for the implementation to be successful. Each milestone consists of one or more action points which can be assigned to members of your team. For example, the **Introduction to Digital Execution with Mendix** milestone consists of action points such as exploring the Mendix Platform, or assigning the administrator of your Mendix app.

> {{< figure src="/attachments/developerportal/compass/invite.png" alt="A screenshot of the Introduction to Digital Execution with Mendix milestone with some action points completed and others in progress" >}}

Reviewer: Wrong image.
```suggestion
{{< figure src="/attachments/developerportal/compass/milestones.png" alt="A screenshot of the Introduction to Digital Execution with Mendix milestone with some action points completed and others in progress" >}}
```

The milestones are defined for you by Mendix, based on our [Digital Execution](https://www.mendix.com/glossary/digital-execution/) framework. They do not keep track of development milestones - their role is to walk you through the steps which you must complete before development can start.

{{< figure src="/attachments/developerportal/compass/milestones-home.png" alt="The Milestones page, showing some completed and open milestones" >}}

You can view and manage milestones by performing the following steps:

> 1. Log in to [Compass](https://compass.home.mendix.com/).

Reviewer: Do we need this step every time? Perhaps we can assume that they are already in Compass, since that is what the document is all about?

{{% alert color="info" %}}
As a best practice, it can be helpful to organize a workshop with your team, so that you can brainstorm your low-code vision together.
{{% /alert %}}

### Area of Focus

The **Area of Focus** section lists several topics to plan for and consider as part of your goal-setting. The focus areas are defined by Mendix based on the [5 Ps of digital transformation](https://www.mendix.com/glossary/digital-execution/#what-are-the-5-ps-of-digital-transformation). You can define your own measurable key results for each of those areas.

{{% alert color="info" %}}
As a best practice, key results should be granular and measurable in the short term.
{{% /alert %}}

The following focus areas are available:

* **People** - Define goals associated with your team. For example, you may decide that you need to train five Mendix developers within the next three months.
* **Platform** - Set goals to integrate Mendix into your software development lifecycle.
* **Portfolio** - Select the offerings where low-code can bring high value to your organization.
* **Process** - Define goals associated with your business processes. For example, you may decide to implement scrum.

> * **Promotion** - Define goals to help you celebrate your successes and motivate your team to build with Mendix.

Reviewer: I think we need to be clearer that this is a fixed list:
```suggestion
The **Area of Focus** section lists the following topics to plan for and consider as part of your goal-setting.

* **People** - Goals associated with your team. For example, you may decide that you need to train five Mendix developers within the next three months.
* **Platform** - Goals to integrate Mendix into your software development lifecycle.
* **Portfolio** - The offerings where low-code can bring high value to your organization.
* **Process** - Goals associated with your business processes. For example, you may decide to implement scrum.
* **Promotion** - Goals to help you celebrate your successes and motivate your team to build with Mendix.

These focus areas are based on the [5 Ps of digital transformation](https://www.mendix.com/glossary/digital-execution/#what-are-the-5-ps-of-digital-transformation). You can define your own measurable key results for each of those areas.

{{% alert color="info" %}}
As a best practice, key results should be granular and measurable in the short term.
{{% /alert %}}
```

#### Managing your Key Results

To give your team transparency into the progress of your Mendix implementation, make sure to track the progress of your key results. You can do this by editing the value in the **Progress** field of a key result.

{{< figure src="/attachments/developerportal/compass/key-result.png" alt="Tracking the progress of a key result" >}}

## Read More

> * [Portfolio Management](/developerportal/portfolio-management/) - Another Digital Execution tool, geared towards pre-implementation strategic planning

Reviewer: Should **Portfolio Management** and **Compass** be grouped together under "**Digital Execution Tools**" to make it clearer what they are about and who should be using them?

Author: Makes sense, but I'm gonna postpone that to a different PR where I'll also reorganize the release notes for portfolio management
